### PR TITLE
Add metrics for more types of failures

### PR DESCRIPTION
**Pull request by jmorrell**
Analysis of build logs gave some more insight into  some common errors that customers are hitting during builds. Let's measure how frequently these happen.

### CODE REVIEW

File: lib/failure.sh [hunk 1]
```diff
@@ -118,7 +120,7 @@ fail_multiple_lockfiles() {
 
 fail_yarn_lockfile_outdated() {
   local log_file="$1"
-  if grep -qi 'error Your lockfile needs to be updated' "$log_file"; then
+  if grep -qi 'Your lockfile needs to be updated' "$log_file"; then
```
Comment (jmorrell) [inline lib/failure.sh]: Yarn slightly updated their message for this error in the last version

File: lib/failure.sh [hunk 2]
```diff
@@ -245,6 +247,55 @@ fail_invalid_semver() {
...
+
+  if grep -qi "Failed at the bcrypt@\d.\d.\d install script" "$log_file"; then
+    mcount "failures.bcrypt-permissions-issue"
+  fi
+
+  if grep -qi "Versions of @angular/compiler-cli and typescript could not be determined" "$log_file"; then
+    mcount "failures.ng-cli-version-issue"
+  fi
+
+  if grep -qi "Versions of @angular/compiler-cli and typescript could not be determined" "$log_file"; then
```
Comment (edmorley) [inline lib/failure.sh]: This check is a duplicate of the one above?

Comment (dzuelke) [reply lib/failure.sh]: Yeah, duplicate here!

Comment (jmorrell) [reply lib/failure.sh]: Nice catch @edmorley! 😅

**Pull request merged**